base/strings/utf8proc.jl

```diff
@@ -18,7 +18,7 @@ export normalize_string, graphemes, is_assigned_char, textwidth, isvalid,
     isvalid(value) -> Bool
 
 Returns `true` if the given value is valid for its type, which currently can be either
-`Char` or `String`.
+`Char` or `String` or `SubString` (it is considered a valid `String` type here).
```

Comment:
or `SubString{String}`

Reply:
I don't understand what "(it is considered a valid `String` type here)" means. `SubString` is not a `String`.

Reply:
I agree that `SubString` is not a `String` but what I meant to say here was that here we get a `true` on `isvalid()` for `SubString` because we are kind of treating them as String of sorts. I guess the wording is a little misleading but just saying "`Char` or `String` or `SubString{String}`." would be a little inappropriate too in my opinion because the entire reason of the issue and this pull request was that `SubString{String}` should be treated as valid in `isvalid`.

Reply:
There's indeed something weird with `isvalid(String, ss::SubString)`. Maybe it should be `isvalid(AbstractString, ss::SubString)`. Then it would be more natural for `SubString` to be a supported type.

Reply:
Yeah I agree or else if `SubString` was a subtype of `String` it would make more sense. I wrote the docs thinking something along those lines to explain it.

Reply:
The doc should be `SubString{String}` as I have suggested above. Additionally the code should remain `String` not `AbstractString`.

Actually the documentation it is good as:
```
 Returns `true` if the given value is valid for its type, which currently can be either
`Char` or `String` or `SubString{String}`.
```

Why? We are checking if `s::SubString{String}` is valid for its type (this is what documentation states) and it is valid if `String(s)` would be a valid `String` as it is a substing of `String`.

Reply:
That makes sense.